src/engine/sync/synccontrol.cpp

```diff
@@ -129,7 +127,21 @@ void SyncControl::setSyncMode(SyncMode mode) {
     // requires. Bypass confirmation by using setAndConfirm.
     m_pSyncMode->setAndConfirm(mode);
     m_pSyncEnabled->setAndConfirm(mode != SYNC_NONE);
-    m_pSyncMasterEnabled->setAndConfirm(isMaster(mode));
```

Comment:
the same is also in InternalClock::setSyncMode(SyncMode mode)

Reply:
Mm I am actually unsure, because a toggle attempt sets it directly to Explicite Master. I am not sure if this is not to much magic if we consider a script user. Please add a comment at the constructor that describes the special case here.

Reply:
sync_master as a light wants to show the type of master: off, soft, or explicit.  But as a button, it only works like this:

* if currently off, requests explicit mode
* if currently soft, requests explicit mode
* if currently explicit, changes to soft mode

so it's still a three state button, but the user can't just toggle through all the settings as usual.

Reply:
> The user offset is lost if you play one deck explicit master enabled with another deck and then switch master to the other deck. Expected behavior: The phase offset adusted by rate_temp_up/down shall remain when switching master

oooo that's tough, let me think about how to handle that

Reply:
This took a lot to fix.  The code was not properly considering user offsets in a bunch of circumstances.  It should work better now.